File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* twoSum(int* numbers, int numbersSize, int target, int* returnSize) {

    int first=0;
    int last=numbersSize-1;
    int sum=0;
    *returnSize=0;
    while(first<last)
    {
        sum= numbers[first]+numbers[last];
        printf("%d...%d===", sum,target);
        if(sum<target)
        {
            first++;
        }
        else if(sum>target)
            last--;
        else
        {
            int *ret= (int *)malloc(sizeof(int)*2);
            ret[0]=first+1;
            ret[1]=last+1;
            *returnSize=2;
            return ret;
        }
    }
    return NULL;
}

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, nums: List[int], target: int):

        left=0 
        right=len(nums)-1
        
        while left < right:
            sum = nums[left]+nums[right]
            if sum < target:
                left +=1
            elif sum > target:
                right -=1
            else:
                return [left+1, right+1]

        return []

        